data-serving/data-service/src/controllers/preprocessor.ts

```diff
@@ -116,22 +116,25 @@ export const setBatchUpsertRevisionMetadata = async (
                         c.caseReference.sourceEntryId,
                 );
                 if (existingCase !== undefined && existingCase.equalsJSON(c)) {
-                    unchangedIds.add(existingCase._id);
+                    unchangedIds.add(existingCase._id.toString());
                 }
             }
             return unchangedIds;
         },
         new Set(),
     );
 
+    // Store the unchanged IDs for future middleware.
+    response.locals.unchangedCaseIdSet = unchangedCaseIdSet;
```

Comment:
nit, we're not using views so putting locals in request.locals feels a bit more appropriate but no big deal either way

Reply:
For some reason, in the Express dts (`express-serve-static-core`), `locals` is only present on Response and Application. /shrug